pages/instructorAssessmentQuestionStatistics/instructorAssessmentQuestionStatistics.js

```diff
@@ -29,20 +30,26 @@ router.get('/', function (req, res, next) {
   setFilenames(res.locals);
   async.series(
     [
-      function (callback) {
-        var params = { assessment_id: res.locals.assessment.id };
-        sqldb.queryOneRow(sql.assessment_stats_last_updated, params, function (err, result) {
-          if (ERR(err, callback)) return;
-          res.locals.stats_last_updated = result.rows[0].stats_last_updated;
-          callback(null);
+      async () => {
+        // make sure statistics are up to date
+        await assessment.updateAssessmentStatistics(res.locals.assessment.id);
+
+        // re-fetch assessment to get updated statistics
+        const result = await sqldb.queryOneRowAsync(sql.select_assessment, {
+          assessment_id: res.locals.assessment.id,
         });
+        res.locals.assessment = result.rows[0].assessment;
       },
       function (callback) {
-        debug('query assessment_stats');
+        // Fetch assessments.stats_last_updated (the time when we last updated
+        // the _question_ statistics for this assessment). Note that this is
+        // different to assessments.statistics_last_updated_at (the time we last
+        // updated the assessment instance statistics stored in the assessments
+        // row itself).
         var params = { assessment_id: res.locals.assessment.id };
-        sqldb.queryOneRow(sql.assessment_stats, params, function (err, result) {
+        sqldb.queryOneRow(sql.assessment_stats_last_updated, params, function (err, result) {
```

Comment:
Nit: while you're here, you could convert this to an async call.

Reply:
I converted the entire file to async/await in 49ea0f5003de058775a03834c00a05546602240a